pkg/controllers/crds/crds_controller.go

```diff
@@ -58,6 +58,7 @@ const (
 
 type Reconciler struct {
 	client.Client
+	isLeader        *bool
```

Comment:
I don't think this bool needs to be a pointer, right? 🤔

Reply:
We do have to pass it by reference so we're able to modify it from the outside (the other goroutine sets it to `true` eventually).

Reply:
Ah I see now, sorry. Why not pass in the channel and then block on reading from it. Once read, set an internal boolean to `true` and then you don't have to bother with passing around the bool by reference. WDYT?

Reply:
Also you can make it none blocking via a select and returning on default.